services/streammanager/kafka/kafkamanager.go

```diff
@@ -248,6 +248,24 @@ func NewProducer(destConfigJSON interface{}, o Opts) (*producerImpl, error) { //
 		return nil, fmt.Errorf("invalid configuration: %w", err)
 	}
 
+	convertToAvro := destConfig.ConvertToAvro
+	avroSchemas := destConfig.AvroSchemas
+	var codecs map[string]goavro.Codec
+	if convertToAvro {
+		codecs = make(map[string]goavro.Codec, len(avroSchemas))
+		for i, avroSchema := range avroSchemas {
+			if len(avroSchema.SchemaId) > 0 {
+				newCodec, err := goavro.NewCodec(avroSchema.Schema)
+				if err != nil {
+					return nil, fmt.Errorf("unable to create codec for schemaId:%+v, with error: %w", avroSchema.SchemaId, err)
+				}
+				codecs[avroSchema.SchemaId] = *newCodec
```

Comment:
How come you're storing it by value here?

Reply:
I didn't get your question.

Reply:
So you are asking me to declare `codecs` as a map of `goavro.Codec` type pointer instead of values? I don't know how to do that.

Reply:
```go
codecs = make(map[string]*goavro.Codec, len(avroSchemas))
```